pkg/bpf/bpffs.go

```diff
@@ -258,7 +258,7 @@ func checkOrMountDefaultLocations() error {
 			return err
 		}
 		if !cMounted {
-			err = mountFS()
+			err = mountFS() // ineffassign: ignore
```

Comment:
Do you think that this ineffassign is because of the `... err := isBPFFS()` on line 256?

Reply:
That might be it. Trying this out https://play.golang.org/p/S8liQ6Y4UsM without the if produced output as expected. I had thought assignments like this didn't shadow already existing variables unless it was in the `if` setup section (`if z=1; z < 9`).

I think I would prefer to refactor this function a little. Either use a big switch where each case corresponds to the various combinations of `mounted` and `bpffsInstance` or return early when possible. So, an early return on line 231, another early return when `mounted && bpffsInstance`, the implied default here. I'm not sure what the precedence is between `!mounted` and `!bpffsInstance` is  but both of those could return internally directly, since they are simple enough.

Reply:
Yes, you are right. I also didn't expect that `v, err := something()` shadows the previously declared `err` variable, but it seems tat it does. I will early return in the separate PR.